### ISSUE TITLE
SmvCsvStringData skips the wrong line without warning

### ISSUE THREAD

**Issue by XiaosenChen** [TresAmigosSD/SMV#733]
When SmvCsvStringData encourters a line that is the wrong length. The behavior is to skip this line without warning. This can be dangerous. It think it should either fail or produce an warning.

**Comment by ninjapapa** [comment]
Can't reproduce. @XiaosenChen could you share an example?

**Comment by laneb** [comment]
Are you still experiencing this problem?

**Comment by XiaosenChen** [comment]
Here are some examples (version 1.5.2.9):
```
class Test1(SmvCsvStringData):
    def schemaStr(self): 
        return "drug:String; showOP:Boolean"

    def dataStr(self):
        return """WEL, True"""

class Test2(SmvCsvStringData):
    def schemaStr(self): 
        return "drug:String; showOP:Boolean"

    def dataStr(self):
        return """WEL, True;
                  Dec,True"""

class Test3(SmvCsvStringData):
    def schemaStr(self): 
        return "drug:String; showOP:Boolean"

    def dataStr(self):
        return """WEL, True; 
                  Dec,1,True"""
```
test1 and test 3 results in empty data frame
test2 has a df with one record: Dec, True
None cause error

**Comment by XiaosenChen** [comment]
It seems:
1) when a record has a space in it, the record gets skipped
2) when a record is the wrong length, the record gets skipped

**Comment by ninjapapa** [comment]
The root cause of this issue is that on Python side `SmvCsvStringData` didn't mix in `WithParser`. It also explains the behavior difference between `SmvCsvStringData` and CreateDF, which purely uses Scala side `SmvCsvStringData`.

**Comment by ninjapapa** [comment]
Fixed by PR #836, #837
